instrumentation/github.com/labstack/echo/otelecho/echo.go

```diff
@@ -15,89 +15,60 @@
 package otelecho // import "go.opentelemetry.io/contrib/instrumentation/github.com/labstack/echo/otelecho"
 
 import (
-	"fmt"
+	"context"
+	"net/http"
 
 	"github.com/labstack/echo/v4"
-	"github.com/labstack/echo/v4/middleware"
-
-	"go.opentelemetry.io/otel"
 
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
 	"go.opentelemetry.io/otel/attribute"
-	"go.opentelemetry.io/otel/propagation"
-	semconv "go.opentelemetry.io/otel/semconv/v1.10.0"
-	oteltrace "go.opentelemetry.io/otel/trace"
+	"go.opentelemetry.io/otel/trace"
 )
 
+type echoCtxKey string
+
 const (
-	tracerKey  = "otel-go-contrib-tracer-labstack-echo"
-	tracerName = "go.opentelemetry.io/contrib/instrumentation/github.com/labstack/echo/otelecho"
+	echoPathCtxKey echoCtxKey = "echoPathCtxKey"
 )
 
 // Middleware returns echo middleware which will trace incoming requests.
-func Middleware(service string, opts ...Option) echo.MiddlewareFunc {
-	cfg := config{}
-	for _, opt := range opts {
-		opt.apply(&cfg)
-	}
-	if cfg.TracerProvider == nil {
-		cfg.TracerProvider = otel.GetTracerProvider()
-	}
-	tracer := cfg.TracerProvider.Tracer(
-		tracerName,
-		oteltrace.WithInstrumentationVersion(SemVersion()),
-	)
-	if cfg.Propagators == nil {
-		cfg.Propagators = otel.GetTextMapPropagator()
-	}
+func Middleware(service string, opts ...otelhttp.Option) echo.MiddlewareFunc {
```

Comment:
What if we would like to have an "echo"-specific option?

Maybe better to use the approach that was proposed by @MadVikingGod some time ago: https://github.com/open-telemetry/opentelemetry-go-contrib/issues/746#issuecomment-821660165?

Reply:
@pellared I've added an option within the otelecho pkg to set otelhttp options. This would allow for any "echo" specific options to be added in the future. I went this way rather than mirroring otelhttp options as that doesn't seem to scale well and makes upgrades a pain. However the user experience of adding an otelhttp option isn't great i feel the benefit outweighs that con. Do you have any suggestions here?

Since otelhttp is locked to a version it should be easy to upgrade the package as is necessary.

Reply:
>  I feel the benefit outweighs that con

It was criticized e.g. here https://github.com/open-telemetry/opentelemetry-go-contrib/issues/746#issuecomment-820402688

Reply:
@pellared can you please indicate the changes you think should be made here?

Reply:
Instead of having `func WithOTelHTTPOptions(options ...otelhttp.Option) Option {` we should have functions like `WithPropagators`, `WithMeterProvider`, `WithMessageEvents`

Reference: https://github.com/open-telemetry/opentelemetry-go-contrib/issues/746#issuecomment-821235264

Reply:
@pellared sorry for the delay. i've just pushed a change to mirror the otelhttp options in otelecho now